tests/test_urimatch.py

```diff
@@ -0,0 +1,35 @@
+
+
+import re
+
+from pytest_httpserver import HTTPServer, URIPattern
+import requests
+
+
+class PrefixMatch(URIPattern):
+    def __init__(self, prefix: str):
+        self.prefix = prefix
+
+    def match(self, uri):
+        return uri.startswith(self.prefix)
+
+
+def test_uripattern_object(httpserver: HTTPServer):
+    httpserver.expect_request(PrefixMatch("/foo")).respond_with_json({"foo": "bar"})
+    assert requests.get(httpserver.url_for("/foo")).json() == {"foo": "bar"}
+    assert requests.get(httpserver.url_for("/foobar")).json() == {"foo": "bar"}
+    assert requests.get(httpserver.url_for("/foobaz")).json() == {"foo": "bar"}
+
+    assert requests.get(httpserver.url_for("/barfoo")).status_code == 500
```

Comment:
Shouldn't this actually be a 404?

Reply:
This means that the server has received an unexpected request and when I created the library I thought that it should be 500 and not 4xx as in this case the client did something not correctly and it should be the proper status (the client may handle the 404, but not the 500).

However, I would implement it again, probably 404 would be more correct as you suggested. But I don't want to break the API.